extra/nrnivmodl_core_makefile.in

```diff
@@ -119,22 +118,26 @@ else
 	mod_files_for_cpp_backend = $(mod_all_files)
 endif
 
-# CPP files and their obkects
-mod_cpp_files = $(patsubst %.mod,%.cpp,$(mod_files_for_cpp_backend))
-mod_cpp_objs = $(addprefix $(MOD_OBJS_DIR)/,$(addsuffix .o,$(basename $(mod_files_no_ext))))
+# CPP files and their objects
+# This is the list of cpp files compile by this make script
+# but in case of ispc, others cpp files will be produced by nmodl itself
+mod_cpp_files = $(mod_files_for_cpp_backend:.mod=.cpp)
+# It's why we take all of them here
+mod_cpp_objs = $(addprefix $(MOD_OBJS_DIR)/,$(mod_files_name:.mod=.o))
 
 # For ISPC backend, we use all non-artificial cells
 mod_ispc_srcs_names = $(notdir $(mod_non_art_files))
-mod_ispc_files = $(patsubst %.mod,%.ispc,$(mod_non_art_files))
-mod_ispc_cpp_files = $(patsubst %.mod,%.cpp,$(mod_non_art_files))
-mod_ispc_objs = $(addprefix $(MOD_OBJS_DIR)/,$(addsuffix .obj,$(basename $(mod_ispc_srcs_names))))
+mod_ispc_files = $(mod_non_art_files:.mod=.ispc)
```

Comment:
I find this syntax in the documentation. They told to use that instead of patsubst if you only want to change the ending